state/src/storage/tx/matching_pools.rs

```diff
@@ -95,11 +82,6 @@ impl<'db> StateTxn<'db, RW> {
 
     /// Destroy a matching pool
     pub fn destroy_matching_pool(&self, pool_name: &str) -> Result<(), StorageError> {
-        // Check that the pool is empty
-        if !self.matching_pool_is_empty(pool_name)? {
-            return Err(StorageError::Other(MATCHING_POOL_NOT_EMPTY_ERR.to_string()));
-        }
-
         let all_pools_key = all_matching_pools_key();
```

Comment:
should we remove the pool mapping of any orders left in the pool when it is destroyed? That way, an order is still matchable even if its pool is destroyed.

Reply:
Oh yikes yea good catch, totally blanked on this. That way orders actually get "moved" to the global pool